documentation/src/main/asciidoc/topics/contributing/terminology.adoc

```diff
@@ -0,0 +1,86 @@
+capabilities and features. It is important to use consistent terminology across
+documentation and the code base to avoid unclear or contradictory language.
+
+Add::
+Add an item to an existing list, group, view, or other container element. For example, "add entries to caches".
+
+Cache configuration::
+An XML or JSON document that controls {brandname} behavior.
+
+Cache template::
+An artifact from which you create cache configurations. +
+{brandname} provides some pre-defined templates that can be referenced using the `org.infinispan.client.hotrod.DefaultTemplate` enum. +
+Users can define new templates in `infinispan.xml` with the `*-cache-configuration` tags. +
+In some APIs, you can use the configuration of an existing cache as a template to configure a new cache. +
+Users can also define templates in `infinispan.xml` with the `*-cache-configuration` tags.
+
+Cache definition::
+Do not use to mean "cache configuration".
+
+Cache Manager::
+Interface for creating and managing caches. Always spell as two words with capital letters when referring to the abstract notion of a Cache Manager. +
+When referring to specific interfaces, use `CacheManager`, `EmbeddedCacheManager`, or `RemoteCacheManager`.
+
+Clear::
+Delete all the elements of a container. For example, "clear entries in a cache".
+
+Create::
+Build something new. For example, "create a cache".
+
+Delete::
+Remove a single element or a group of similar elements from a container. For example, "delete an entry from the cache" or "delete all expired entries from the cache".
+
+Destroy::
+Do not use to mean "delete".
+
+Index::
+Create an index to improve query performance or add an entry to an existing index when it is added to a cache.
+
+Indexes::
+Plural of index. Do not use "indices".
+
+Purge::
+Can have negative connotations and is not a preferred term. Use only as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
+
+Query::
+Look up information within the data set. The words "query" and "search" are synonymous but frequently interchanging the two can lead to inconsistency.
++
+Use in verb form: "Query values in the cache." +
+Use in noun form preceded by Ickle: "Perform an Ickle query on the cache."
+
+Reindex::
+Rebuild an index to update or change an existing index. Do not hyphenate (always "reindex" never "re-index").
+
+Remove::
+Use as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
+
+Reset::
+Restore an initial value or state. For example, "Restore a counter."
+
+Search::
+Synonymous with "query" but not preferred. In verb form, use with common phrases like "full-text search" in conjunction with {brandname} Query API. In noun form when referring to the "org.infinispan.query.Search" entry point.
```

Comment:
I would put `org.infinispan.query.Search` in monospace font instead of using quotes.